docs/understand/texture_fetching.rst

```diff
@@ -0,0 +1,107 @@
+
+Using float indices isn't trivial. The two issues that come up are sampling and addressing.
+
+When a texture is indexed with a fraction, the queried value is often between two or more texels (texture elements). The sampling method defines what value to return in these cases.
+
+Sometimes the index is outside the bounds of the texture. This might seem unintuitive, but is useful in computer graphics for putting a texture on a surface multiple times, or just creating a visible sign of out of bounds indexing. Addressing mode defines what value to return when indexing a texture out of bounds.
+
+There are several different sampling and addressing modes, and no rule for how to choose between them, as it is dependent on the application. This means that it has to be decided during development which sampling and addressing method to use.
+
+This image is the texture, that will be used as for the examples below. It is 2x2 texels and is indexed in the [0..1] range.
+
+.. image:: ../data/understand/textures/original.png
+  :width: 150
+  :alt: Texture used as example
+
+
+.. _texture_fetching_nearest:
+Nearest point sampling
+===============================================================================
+
+In this mode the ``tex(x) = T[floor(x)]`` and similarly for 2D and 3D variants. In practice this will not interpolate between neighboring values, which results in a pixelated look.
+
+This image is the example texture stretched out to a 4x4 pixel quad, but still indexed in the [0..1] range. The in between values are the same as the values of the nearest texel.
+
+.. image:: ../data/understand/textures/nearest.png
+  :width: 300
+  :alt: Texture upscaled with nearest point sampling
+
+.. _texture_fetching_linear:
+Linear filtering
+===============================================================================
+
+Linear filtering method simply does a linear interpolation between values.
+
+* For one dimensional textures it is ``tex(x) = (1-α)T[i] + αT[i+1]``
+* For two dimensional textures it is ``tex(x,y) = (1-α)(1-β)T[i,j] + α(1-β)T[i+1,j] + (1-α)βT[i,j+1] + αβT[i+1,j+1]``
+* For three dimensional textures it is ``tex(x,y,z) = (1-α)(1-β)(1-γ)T[i,j,k] + α(1-β)(1-γ)T[i+1,j,k] + (1-α)β(1-γ)T[i,j+1,k] + αβ(1-γ)T[i+1,j+1,k] + (1-α)(1-β)γT[i,j,k+1] + α(1-β)γT[i+1,j,k+1] + (1-α)βγT[i,j+1,k+1] + αβγT[i+1,j+1,k+1]``
+
+Where ``i = round(x')``, ``α = frac(x')``, ``x' = x - 0.5``, ``j = round(y')``, ``β = frac(y')``, ``y' = y - 0.5``, ``k = round(z')``, ``γ = frac(z')`` and ``z' = z - 0.5``
```

Comment:
This could need a better explanation, that does more than just listing the equations. Or at least explain the variables for the one dimensional case, so that the others are easier to follow